lib/orm2-adapter.js

```diff
@@ -87,35 +92,35 @@ export default class NodeORM2Adapter extends ORMAdapter {
   }
 
   saveRecord(model) {
-    return fromNode(model.record.save.bind(model.record));
+    return fromNode((cb) => model.record.save(cb));
   }
 
   deleteRecord(model) {
-    return fromNode(model.record.remove.bind(model.record));
+    return fromNode((cb) => model.record.remove(cb));
   }
 
   defineModels(models) {
-    // Don't define abstract base classes
-    models = models.filter((Model) => !(Model.hasOwnProperty('abstract') && Model.abstract));
-
+    let db = this.container.lookup('database:orm2');
     this.ormModels = {};
 
     // Define models
     models.forEach((Model) => {
       let attributes = {};
-      Model.eachAttribute((key, attribute) => {
+      let modelType = this.container.metaFor(Model).containerName;
```

Comment:
Latest denali version introduces `Model.getType(container)` static method, which is a convenience shortcut for this.